api/v1beta2/ocirepository_types.go

```diff
@@ -190,6 +190,26 @@ type OCIRepositoryVerification struct {
 	// trusted public keys.
 	// +optional
 	SecretRef *meta.LocalObjectReference `json:"secretRef,omitempty"`
+
+	// CosignIdentityMatch specifies the identity matching criteria to use
+	// while verifying an OCI artifact which was signed using Cosign keyless
+	// signing.
+	CosignIdentityMatch *CosignIdentityMatch `json:"cosignIdentityMatch,omitempty"`
```

Comment:
I propose the following changes to the API:

```yaml
verify:
  provider: cosign
  matchIdentity:
    issuer: "<regexp>"
    subject: "<regexp>"
```

The match would then be common to any other OIDC-based signing tools like OpenPubkey.

Reply:
the reason i added "cosign" as a prefix because i was not sure what naming other tools might use. but since "issuer" and "subject" are related to OIDC, any OIDC-based signing tool could make use of these fields. so i propose we change `matchIdentity` to `oidcMatchIdentity`, so that we can shield this field from signing tools that don't use OIDC.

Reply:
The verb should go first `matchOIDCIdentity`